### PR TITLE
[llvmonly] Mixed mode changes.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/ee.h [hunk 1]
```diff
@@ -30,6 +30,7 @@ struct _MonoEECallbacks {
 	void (*entry_from_trampoline) (gpointer ccontext, gpointer imethod);
 	void (*to_native_trampoline) (gpointer addr, gpointer ccontext);
 	gpointer (*create_method_pointer) (MonoMethod *method, gboolean compile, MonoError *error);
+	MonoFtnDesc *(*create_method_pointer_llvmonly) (MonoMethod *method, gboolean compile, MonoError *error);
```
Comment (lewurm) [inline mono/mini/ee.h]: needs API version bumping

File: mono/mini/ee.h [hunk 2]
```diff
@@ -30,6 +30,7 @@ struct _MonoEECallbacks {
 	void (*entry_from_trampoline) (gpointer ccontext, gpointer imethod);
 	void (*to_native_trampoline) (gpointer addr, gpointer ccontext);
 	gpointer (*create_method_pointer) (MonoMethod *method, gboolean compile, MonoError *error);
+	MonoFtnDesc *(*create_method_pointer_llvmonly) (MonoMethod *method, gboolean unbox, MonoError *error);
```
Comment (lewurm) [inline mono/mini/ee.h]: needs API version bumping

File: mono/mini/interp/interp.c [hunk 3]
```diff
@@ -2513,6 +2519,87 @@ no_llvmonly_interp_method_pointer (void)
...
+			entry_func = entry_funcs_static [sig->param_count];
+		else
+			entry_func = entry_funcs_static_ret [sig->param_count];
+	}
+	g_assert (entry_func);
+
+	/* Encode unbox in the lower bit of imethod */
+	gpointer entry_arg = imethod;
+	if (unbox)
+		entry_arg = (gpointer)(((gsize)entry_arg) | 1);
```
Comment (lewurm) [inline mono/mini/interp/interp.c]: can we add a field to `InterpMethod` instead`?

File: mono/mini/interp/interp.c [hunk 4]
```diff
@@ -2513,6 +2519,87 @@ no_llvmonly_interp_method_pointer (void)
...
+	 * The entry functions need access to the method to call, so we have
+	 * to use a ftndesc. The caller uses a normal signature, while the
+	 * entry functions use a gsharedvt_in signature, so wrap the entry function in
+	 * a gsharedvt_in_sig wrapper.
+	 */
+	wrapper = mini_get_gsharedvt_in_sig_wrapper (sig);
+
+	entry_wrapper = mono_jit_compile_method_jit_only (wrapper, error);
+	mono_error_assertf_ok (error, "couldn't compile wrapper \"%s\" for \"%s\"",
+			mono_method_get_name_full (wrapper, TRUE, TRUE, MONO_TYPE_NAME_FORMAT_IL),
```
Comment (lewurm) [inline mono/mini/interp/interp.c]: nit: `mono_method_get_full_name (wrapper)` would do the same

File: mono/mini/interp/interp.c (hunk 3 above)
Comment (vargaz) [reply mono/mini/interp/interp.c]: The same method can be called with or without boxing.

**Pull request merged**